mne/io/egi/egimff.py

```diff
@@ -686,3 +692,199 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+    if not fname.endswith('.mff'):
+        raise ValueError('fname must be an MFF file with extension ".mff".')
+    # Confirm the input MFF is averaged
+    mff = mffpy.Reader(fname)
+    if mff.flavor != 'averaged':
+        raise ValueError('%s is a %s MFF file. fname must be the path to an \
+                         averaged MFF file.' % (fname, mff.flavor))
+    # Check for categories.xml file
+    if 'categories.xml' not in mff.directory.listdir():
+        raise ValueError('categories.xml not found in MFF directory. \
+                         %s may not be an averaged MFF file.' % fname)
+    if condition is None:
+        categories = mff.categories.categories
+        condition = list(categories.keys())
+    elif not isinstance(condition, list):
+        condition = [condition]
+    logger.info('Reading %d evoked datasets from %s ...'
+                % (len(condition), fname))
+    output = [_read_evoked_mff(fname, c, channel_naming=channel_naming,
+                               verbose=verbose).apply_baseline(baseline)
+              for c in condition]
+    return output if len(output) > 1 else output[0]
+
+
+def _read_evoked_mff(fname, condition, channel_naming='E%d', verbose=None):
+    """Read evoked data from MFF file."""
+    import mffpy
+    egi_info = _read_header(fname)
+    mff = mffpy.Reader(fname)
+    categories = mff.categories.categories
+
+    if isinstance(condition, str):
+        # Condition is interpreted as category name
+        category = condition
+        epoch = mff.epochs[category]
+    elif isinstance(condition, int):
+        # Condition is interpreted as epoch index
+        epoch = mff.epochs[condition]
+        category = epoch.name
+    else:
+        raise TypeError('Condition must be either int or str.')
+    assert category in categories.keys(), 'Condition "%s" not found in \
+                                           available conditions %s.'\
+                                           % (category, categories.keys())
+
+    # Read in signals from the target epoch
+    data = mff.get_physical_samples_from_epoch(epoch)
+    eeg_data, t0 = data['EEG']
+    if 'PNSData' in data.keys():
+        pns_data, t0 = data['PNSData']
+        all_data = np.vstack((eeg_data, pns_data))
+        ch_types = egi_info['chan_type'] + egi_info['pns_types']
+    else:
+        all_data = eeg_data
+        ch_types = egi_info['chan_type']
+    all_data *= 1e-6  # convert to volts
+
+    # Load metadata into info object
+    # Exclude info['meas_date'] because record time info in
+    # averaged MFF is the time of the averaging, not true record time.
+    ch_names = [channel_naming % (i + 1) for i in
+                range(mff.num_channels['EEG'])]
+    ch_names.extend(egi_info['pns_names'])
+    info = create_info(ch_names, mff.sampling_rates['EEG'], ch_types)
+    info['nchan'] = sum(mff.num_channels.values())
+    assert info['nchan'] == all_data.shape[0], 'Number of channels does \
```

Comment:
Same statement about `assert` as above

Reply:
I don't think this `assert` is doing a lot anyway since we ultimately get the value for both sides of the `==` from the binary signals file, so I vote we just take it out.